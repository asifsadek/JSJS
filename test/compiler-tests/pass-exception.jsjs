val divide = /\(numer: num, denom: num): num => {
  if denom == 0 then throw "divide by zero"
  else numer / denom;
};


val answer = /\(): unit => {
  print_string("ello");
  try {divide(10, 0); (-);} catch(e) {
    if e == "divide by zero"
    then print("non zero pls")
    else (-);
  };
};

val something = try {
  val x = 10 + 10;
  throw "try throwing an error";
} catch(e) {
  print(e);
};

answer();
