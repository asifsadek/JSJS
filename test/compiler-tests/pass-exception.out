try throwing an error
non zero pls
